### PR TITLE
Added provision watcher client

### PR DESCRIPTION
**Pull request by jduranf**
Implements #1098 

Signed-off-by: Joan Duran <jduran@circutor.com>

### CODE REVIEW

File: pkg/clients/metadata/provision_watcher.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+	"github.com/edgexfoundry/edgex-go/pkg/models"
+)
+
+/*
+Provision Watcher client for interacting with the provision watcher section of metadata
+*/
+type ProvisionWatcherClient interface {
+	Add(dev *models.ProvisionWatcher, ctx context.Context) (string, error)
+	Delete(id string, ctx context.Context) error
+	ProvisionWatcher(id string, ctx context.Context) (models.ProvisionWatcher, error)
```
Comment (tsconn23) [inline pkg/clients/metadata/provision_watcher.go]: Rename to GetById?

File: pkg/clients/metadata/provision_watcher.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+	ProvisionWatcherForName(name string, ctx context.Context) (models.ProvisionWatcher, error)
```
Comment (tsconn23) [inline pkg/clients/metadata/provision_watcher.go]: Rename to GetByName?

File: pkg/clients/metadata/provision_watcher.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+	ProvisionWatchers(ctx context.Context) ([]models.ProvisionWatcher, error)
```
Comment (tsconn23) [inline pkg/clients/metadata/provision_watcher.go]: Rename to List()?

**Comment by tonyespy** [comment]
Note `ProvisionWatchers` are part of dynamic device discovery, which is currently a stretch goal for the device SDKs in Edinburgh.

**Comment by jduranf** [comment]
@tsconn23 I agree with your comments, but all other clients in [core-data](https://github.com/edgexfoundry/edgex-go/blob/master/pkg/clients/coredata/value_descriptor.go), [core-metadata](https://github.com/edgexfoundry/edgex-go/blob/master/pkg/clients/metadata/device.go) or [scheduler](https://github.com/edgexfoundry/edgex-go/blob/master/pkg/clients/scheduler/interval.go) have the same structure, which I tried to follow. Maybe should be fine to rename all clients to remove the redundant prefix in another PR, but for coherency, in this PR I prefer to keep those names.

**Comment by tsconn23** [comment]
@tonyespy Yes, I had the same thought and considered possibly putting a hold on this PR until the Fuji dev cycle. @jduranf Are you in practical need of this change or are you contributing simply because it's a gap in our functionality?

**Comment by jduranf** [comment]
@tsconn23 I'm in practical need. If you want to add this feature in Fuji, should be great for me to, at least, add the provision watcher client in edgex-go (and maybe the callback system in another PR, but I have to check that).

**Comment by tsconn23** [comment]
@jduranf Depending on the timing of edgex-go's conversion to a module and subsequent import of https://github.com/edgexfoundry/go-mod-core-contracts you might think about making this change in the go-mod-core-contracts repo.

Once go-mod-core-contracts is imported to edgex-go, the /pkg folder in edgex-go will be deleted.

We'll be discussing timing on the Core WG call tomorrow.

**Comment by jduranf** [comment]
This will be done in [go-mod-core-contracts](https://github.com/edgexfoundry/go-mod-core-contracts/pull/6)

**Pull request closed without merging**